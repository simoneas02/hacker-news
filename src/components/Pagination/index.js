import React from 'react';

import pagination from './pagination';

const Pagination = ({ total, activePage, pageLink }) => (
  <ul>
    {pagination({ total, activePage }).map((page, index) => (
      <li key={index}>
        {page === '...' ? (
          <span>{page}</span>
        ) : (
          <a href={pageLink.replace('%page%', page)}>{page}</a>
        )}
        ;
      </li>
    ))}
  </ul>
);

export default Pagination;
