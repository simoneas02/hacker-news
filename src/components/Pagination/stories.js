import React from "react";
import { storiesOf } from "@storybook/react";

import Pagination from ".";

const stories = storiesOf("Pagination", module);

stories.add("without props", () => <Pagination />);

stories.add("with total and activePage", () => (
  <Pagination total={10} activePage={5} />
));

stories.add("with total, activePage and pageLink", () => (
  <Pagination total={3} activePage={1} pageLink="http:mypage.com/page/%page%" />
));

stories.add("with total, activePage, pageLink and callback", () => (
  <Pagination
    total={15}
    activePage={9}
    pageLink="http:mypage.com/page/%page%"
    onClick={page => console.log(page)}
  />
));
