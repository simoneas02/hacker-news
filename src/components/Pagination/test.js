import React from 'react';
import { expect } from 'chai';
import Pagination from '.';

describe('Pagination', () => {
  it('Pagination should be a function', () => {
    expect(Pagination).to.be.a('function');
  });

  test('pagination({total: 1, acitivePage: 1}) should return [1]', () => {
    const params = { total: 1, acitivePage: 1 };
    const result = [1];

    expect(Pagination(params)).to.be.deep.equal(result);
  });

  test('pagination({total: 2, acitivePage: 1}) should return [1, 2]', () => {
    const params = { total: 2, acitivePage: 1 };
    const result = [1, 2];

    expect(Pagination(params)).to.be.deep.equal(result);
  });
});
