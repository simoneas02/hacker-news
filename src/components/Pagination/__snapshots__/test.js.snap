// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination has a valid snapshot 1`] = `
<div>
  Pagination
</div>
`;
